diff --git a/dist/index.js b/dist/index.js
index 69e443eb54d156498994a6272dbc43be220dd3a3..0580c67306cc08302b80415b801f6231175c8de5 100755
--- a/dist/index.js
+++ b/dist/index.js
@@ -848,10 +848,12 @@ var BuildBundler = class extends Bundler {
 	studio;
-	constructor({ studio } = {}) {
+	configEntryFile;
+	constructor({ studio, configEntryFile } = {}) {
 		super("Build");
 		this.studio = studio ?? false;
+		this.configEntryFile = configEntryFile;
 		this.platform = process.versions?.bun ? "neutral" : "node";
 	}
 	async getUserBundlerOptions(mastraEntryFile, outputDirectory) {
-		const bundlerOptions = await super.getUserBundlerOptions(mastraEntryFile, outputDirectory);
+		const bundlerOptions = await super.getUserBundlerOptions(this.configEntryFile ?? mastraEntryFile, outputDirectory);
 		if (!bundlerOptions?.[IS_DEFAULT]) return bundlerOptions;
 		return {
@@ -964,5 +966,8 @@ async function build({ dir, tools, root, studio, debug }) {
 		const platformDeployer = mastraEntryFile ? await getDeployer(mastraEntryFile, outputDirectory) : void 0;
 		if (!platformDeployer) {
-			const deployer = new BuildBundler({ studio });
+			const deployer = new BuildBundler({
+				studio,
+				configEntryFile: mastraEntryFile
+			});
 			deployer.__setLogger(logger);
 			const discoveredTools = deployer.getAllToolPaths(mastraDir, [...tools ?? [], ...fsAgents.toolPaths]);
